docs: update styles docs

rise-erpelding opened the pull request:
<!---
    - Following conventional commit format, provide a general summary of your changes in the title above.
    - Acceptable commit types in order of severity (high to low): feat, fix, docs, style, chore, perf, and test. Commit types are defined in PULL_REQUESTS.md.
    - For example,`type(component): general summary`
-->

## Description

Updates the `@spectrum-web-components/styles` package documentation for clarity and completeness.

**Updated 11/11**: Expands documentation to more thoroughly describe tokens, with code snippets, while minimizing documentation about themes files, which will be deprecated.

## Motivation and context

Part of a larger docs improvement initiative to update documentation.

## Related issue(s)

<!---
    - If suggesting a new feature or change, please discuss it in an issue first.
    - If fixing a bug, include the issue number where the reviewers can find a description of the bug with steps to reproduce.
    - If you're an Adobe employee, add a Jira ticket number but DO NOT LINK directly to Jira.
-->

-   closes SWC-432

---

## Author's checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply.  If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** and **[PULL_REQUESTS](<(https://github.com/adobe/spectrum-web-components/blob/main/PULL_REQUESTS.md)>)** documents.
-   [ ] <strike>I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)</strike>
-   [ ] <strike>I have added automated tests to cover my changes.</strike>
-   [ ] <strike>I have included a well-written changeset if my change needs to be published.</strike>
-   [x] I have included updated documentation if my change required it.

---

## Reviewer's checklist

-   [x] Includes a Github Issue with appropriate flag or Jira ticket number without a link
-   [ ] Includes thoughtfully written changeset if changes suggested include `patch`, `minor`, or `major` features
-   [ ] Automated tests cover all use cases and follow best practices for writing
-   [ ] Validated on all supported browsers
-   [ ] All VRTs are approved before the author can update Golden Hash

### Manual review test cases

<!---
    - For the author, please describe in detail what reviewers should test.
    - Include links and manual steps for how the reviewer should go through to verify your changes.
    - Be sure to include all areas of the codebase that might be affected. Any components that use these changes for a dependency should be cross-checked for regressions.
    - For example, changes to Menu Item will affect Picker, Menu, and Action Menu.
-->

## 1. Link validation
- [ ] All internal links work as expected
- [x] All external links work as expected

## 2. Code examples and imports
- [ ] All import paths reference files that actually exist in the package
- [ ] Code examples are accurate and functional
- [ ] Language tags are correct (CSS imports use `css` not `ts`)

## 3. Terminology consistency
- [ ] Terms like "system variants", "color options", and "scales" are used consistently throughout

## 4. Markdown formatting and readability
- [ ] No duplicate headings
- [ ] Proper heading hierarchy
- [ ] Tables render correctly
- [ ] Sentence case is used consistently


### Device review

<!--- Verify the above manual tests and visual accuracy utilizing an emulator like Polypane browser or on an actual device. -->

-   [ ] Did it pass in Desktop?
-   [ ] Did it pass in (emulated) Mobile?
-   [ ] Did it pass in (emulated) iPad?

Rajdeepc commented on 1st-gen/tools/styles/README.md:
````diff
@@ -1,124 +1,240 @@
+
+> **Note:** The `darkest` and `lightest` colors are deprecated and will be removed in a future release. Use `dark` or `light` instead.
+
+```ts
````
```suggestion
```js
```

Rajdeepc commented on 1st-gen/tools/styles/README.md:
```diff
@@ -1,124 +1,240 @@
+</sp-table>
 
-### Lightest
+## Typography
```
```suggestion
### Typography
```

Rajdeepc commented on 1st-gen/tools/styles/README.md:
````diff
@@ -1,124 +1,240 @@
 
-```
-@import '@spectrum-web-components/styles/theme-lightest.css';
+```ts
````
```suggestion
```js
```

Rajdeepc commented on 1st-gen/tools/styles/README.md:
```diff
@@ -1,124 +1,240 @@
+This file provides the complete [Spectrum Typography system](https://opensource.adobe.com/spectrum-css/?path=/docs/components-typography--docs). The typography system is shared across all design systems (Spectrum Legacy, Express, and Spectrum 2).
 
-## Scale
+**What's included:**
```
```suggestion
**What's included**
```

Rajdeepc commented on 1st-gen/tools/styles/README.md:
```diff
@@ -1,124 +1,240 @@
-## Typography
+Import individual exports when you only need specific typography components to reduce bundle size.
+
+## Design tokens
```
```suggestion
### Design tokens
```

rise-erpelding commented:
@Rajdeepc I think I've addressed most of your feedback here, thanks!

I removed a lot of the references to "Legacy" since it doesn't seem like that's terminology we're using.

I made a lot of adjustments to heading levels, although I ended up getting down to `h5` and those look a little strange to me, happy to change those if we decide we'd rather not use that heading level.

I addressed the syntax highlighting suggestions too! I realized that a lot of these are `@import` syntax which is css, so it's neither Typescript nor JavaScript.

Last thing: you left a comment about code snippets, I haven't addressed this yet but wanted to push up what I had today.

rise-erpelding commented on 1st-gen/tools/styles/package.json:
There were lots of lint warnings about keys being in the wrong order, so I've reordered things, happy to roll that back if that's not what we want!

marissahuysentruyt commented on 1st-gen/tools/styles/README.md:
```diff
@@ -1,124 +1,241 @@
+
+- Use `<sp-theme>` for automatic theme management (recommended)
+- Import Spectrum Core tokens directly for granular control (see "Design tokens" section above)
+- See the [Spectrum Core tokens migration guide](../theme/core-tokens.md) for help migrating from Spectrum Vars to Spectrum Core tokens
```
I don't think this link works. I think this should fix it: 

`tools/core-tokens/`

marissahuysentruyt commented on 1st-gen/tools/styles/README.md:
```diff
@@ -1,124 +1,241 @@
+    - **Colors:** `dark`, `light`
+    - **Scales:** `medium`, `large`
 
-This file brings together the global variables and font settings with the "Light" color set and "Medium" scale system specification.
```
So happy we're removing all of the repetitive stuff 👍

marissahuysentruyt commented on 1st-gen/tools/styles/README.md:
````diff
@@ -1,124 +1,241 @@
-```
-@import '@spectrum-web-components/styles/theme-darkest.css';
-```
+Generally speaking, `light-vars` and `dark-vars` files contain raw color custom property definitions (such as `--spectrum-red-500`), as well as semantic custom property definitions that change depending on which color option you use. For instance, in the v2-tokens, `--spectrum-neutral-subdued-background-color-default` uses different values per theme. It resolves to `--spectrum-gray-700` for light ...
````
Perhaps instead of `v2-tokens`, we use `tokens-v2` since that's the directory we have? 

I'll leave that up to you- not a blocker.

marissahuysentruyt reviewed (approved):
Thanks for making those changes so quickly- looks great. 
![finger guns](https://github.com/user-attachments/assets/75bce036-fc73-4ea4-92c0-35fc7e70b24c)

caseyisonit commented on 1st-gen/tools/styles/README.md:
````diff
@@ -1,124 +1,264 @@
+@import '@spectrum-web-components/styles/tokens-v2/large-vars.css'; /* import scale tokens */
+@import '@spectrum-web-components/styles/tokens-v2/global-vars.css'; /* import global tokens */
+@import '@spectrum-web-components/styles/typography.css';
 ```
````
i agree with this direction, just because someone can, doesnt mean they should. we should only ever document supported patterns. reduce the number of ways something can be done makes it easier to maintain in the long run!

rise-erpelding commented on 1st-gen/tools/styles/README.md:
```diff
@@ -28,27 +28,98 @@ yarn add @spectrum-web-components/styles
+
+Here is a summary of each system and the options available on each:
+
+<sp-table>
```
I added a table. It's a nice way to present information but the information being presented seems kind of obvious, so I took it out. I committed it so I could bring it back if a reviewer feels differently about it though. Here's what it looked like:

<img width="979" height="158" alt="Screenshot 2025-11-20 at 3 26 40 PM" src="https://github.com/user-attachments/assets/838c518f-c1c2-4219-a3a5-138df269dad4" />

caseyisonit reviewed (approved):
LGTM
